diff --git a/src/ts_config.c b/src/ts_config.c
index c9e9554..3b83f07 100644
--- a/src/ts_config.c
+++ b/src/ts_config.c
@@ -130,9 +130,9 @@ static int __ts_config(struct tsdev *ts, char **conffile_modules,
 				printf("TSLIB_CONFFILE: module %s %s\n",
 					module_name, p);
 			#endif
-				sprintf(conffile_modules[line], module_name);
+				sprintf(conffile_modules[line], "%s", module_name);
 				if (conffile_params)
-					sprintf(conffile_params[line], p);
+					sprintf(conffile_params[line], "%s", p);
 			}
 		} else if (strcasecmp(tok, "module_raw") == 0) {
 		#if !defined HAVE_STRSEP
@@ -148,9 +148,9 @@ static int __ts_config(struct tsdev *ts, char **conffile_modules,
 				printf("TSLIB_CONFFILE: module_raw %s %s\n",
 					module_name, p);
 			#endif
-				sprintf(conffile_modules[line], module_name);
+				sprintf(conffile_modules[line], "%s", module_name);
 				if (conffile_params)
-					sprintf(conffile_params[line], p);
+					sprintf(conffile_params[line], "%s", p);
 
 				if (raw)
 					raw[line] = 1;
diff --git a/src/ts_config_filter.c b/src/ts_config_filter.c
index 4eae1d8..d5b5c5a 100644
--- a/src/ts_config_filter.c
+++ b/src/ts_config_filter.c
@@ -89,13 +89,13 @@ struct ts_module_conf *ts_conf_get(struct tsdev *ts)
 		if (!conf_next->name)
 			goto fail;
 
-		sprintf(conf_next->name, modulebuf[i]);
+		sprintf(conf_next->name, "%s", modulebuf[i]);
 
 		conf_next->params = calloc(1, LINE_MAX);
 		if (!conf_next->params)
 			goto fail;
 
-		sprintf(conf_next->params, parambuf[i]);
+		sprintf(conf_next->params, "%s", parambuf[i]);
 
 		conf_next->raw = raw[i];
 
